### ISSUE TITLE
Time.Interval.to_string! not using the `greatest difference` between fields for formatting

### ISSUE THREAD

**Issue by sfusato** [elixir-cldr/cldr_dates_times#21]
Example:
```
iex> Cldr.Time.Interval.to_string! ~T[20:00:00], ~T[22:00:00]
"8:00 – 10:00 PM"

iex> Cldr.Time.Interval.to_string! ~T[08:00:00], ~T[22:00:00]
"8:00 – 10:00 PM"
```
In the 2nd case, I would have expected "8:00 AM – 10:00 PM"

**Comment by kipcole9** [comment]
Bug confirmed and location identified. Working on resolution. Thanks for the report, much appreciated.

**Comment by kipcole9** [comment]
Fixed in  [this commit](https://github.com/elixir-cldr/cldr_dates_times/commit/6e84950a9340d5946dcf5deee821999b7ff43fb1) and published as [ex_cldr_dates_times version 2.7.1](https://hex.pm/pacakges/ex_cldr_dates_times/2.7.1). The changelog entry is:

### Bug Fixes

* Fix formatting time intervals where the start time is "a.m." and the end time is "p.m.". Thanks to @sfusato for the report. Closes #21.

Thanks very much for the collaboration and report.
